pkg/cache/clusterqueue.go

```diff
@@ -454,23 +495,34 @@ func (c *ClusterQueue) reportActiveWorkloads() {
 // and the number of admitted workloads for local queues.
 func (c *ClusterQueue) updateWorkloadUsage(wi *workload.Info, m int64) {
 	admitted := workload.IsAdmitted(wi.Obj)
-	updateUsage(wi, c.Usage, m)
+	updateFlavorUsage(wi, c.Usage, m)
+	updateResourceStats(wi, c.ResourceStats, m)
 	if admitted {
-		updateUsage(wi, c.AdmittedUsage, m)
+		updateFlavorUsage(wi, c.AdmittedUsage, m)
 		c.admittedWorkloadsCount += int(m)
 	}
 	qKey := workload.QueueKey(wi.Obj)
 	if lq, ok := c.localQueues[qKey]; ok {
-		updateUsage(wi, lq.usage, m)
+		updateFlavorUsage(wi, lq.usage, m)
 		lq.reservingWorkloads += int(m)
 		if admitted {
-			updateUsage(wi, lq.admittedUsage, m)
+			updateFlavorUsage(wi, lq.admittedUsage, m)
 			lq.admittedWorkloads += int(m)
 		}
 	}
 }
 
-func updateUsage(wi *workload.Info, flvUsage FlavorResourceQuantities, m int64) {
+func updateResourceStats(wi *workload.Info, rStats ResourceStats, m int64) {
+	for _, ps := range wi.TotalRequests {
+		for res, v := range ps.Requests {
+			if _, exists := rStats[res]; exists {
+				rStats[res].Usage += v * m
```

Comment:
Line 518 already checks this.

Also, rStats should have all the resources that the Workload requests (otherwise the Workload wouldn't have been admitted there).

Reply:
That makes sense. That is a great pointer.
Thank you!